src/ducks/wallet/ledger.ts

```diff
@@ -0,0 +1,70 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import { ActionStatus, RejectMessage } from "constants/types.d";
+import LedgerTransport from "@ledgerhq/hw-transport-u2f";
+import LedgerApi from "@ledgerhq/hw-app-str";
+
+export const fetchLedgerStellarAddressAction = createAsyncThunk<
+  { publicKey: string },
+  string,
+  { rejectValue: RejectMessage }
+>(
+  "walletLedger/fetchLedgerStellarAddressAction",
+  async (bipPath, { rejectWithValue }) => {
+    const result = { publicKey: "" };
+    try {
+      const transport = await LedgerTransport.create();
+      const ledgerApi = new LedgerApi(transport);
+      const response = await ledgerApi.getPublicKey(bipPath);
+      result.publicKey = response.publicKey;
+    } catch (error) {
+      return rejectWithValue({ errorString: error.toString() });
+    }
+    return result;
+  },
+);
+
+interface InitialState {
```

Comment:
I made a change yesterday to have all reducer initial states unique because it's more clear that way in the store. Sorry, I should have mentioned this. 😊